### DISCUSSION TITLE
Image rendering in nvim dashboard

### DISCUSSION THREAD

**Discussion by copeugne** [3rd/image.nvim#24] [Q&A] [answered]
Would it be possible to use this plugin to render an image in the header of an nvim dashboard instead of an ascii art or braille representation of an image ?

![image](https://github.com/3rd/image.nvim/assets/57577919/546caed1-7ff8-435e-bbbb-7662bfab419a)

**Comment by 3rd** [comment]
Yes, definitely!

**Reply by copeugne** [reply]
Amazing !
I'll look into how I can integrate image.nvim inside my alpha.nvim setup. Thank you 😀

**Comment by 3rd** [comment, marked as answer]
@copeugne You'll want to do something like this: https://github.com/3rd/config/blob/master/home/dotfiles/nvim/lua/modules/alpha.lua

It's not perfect, something happens inside alpha that clears the virtual space the plugin injects.
This happend when the window is resized and probably in many other cases.
A workaround would be to disable the virtual text, set a fixed height on the image, and just inject padding from alpha.
Another would be to hook into whatever alpha is doing and call image:render() again after it's done.

Rendering the image might slow down your startup, depending on your computer but also on the image size.
Here's a small demo:


https://github.com/3rd/image.nvim/assets/59587503/f4913bb2-3bef-4abc-8eef-3c1985f45f42

**Reply by copeugne** [reply]
Thank you very much for your help, I wouldn't have been able to get it somewhat working without the code you provided.

I am not on NixOS but on MacOS and I am using LazyVim so I had to tweak it a tiny bit.

https://github.com/3rd/image.nvim/assets/57577919/2de68b24-194f-46e3-9545-dcb56bb16a8e

In the video you can see the image showing up and then disappearing, I don't really get why but when the window is getting resized with my tiling window manager (Yabai) the image is not disappearing anymore for some reason. 
<img width="614" alt="dashboard" src="https://github.com/3rd/image.nvim/assets/57577919/d0bd8d3e-740d-43e9-9f11-13a45c3870c8">

I tried playing around with padding and image size but I didn't really get it to work.

**Reply by copeugne** [reply]
It seems to be working ok with a delay, thank you !

**Reply by 3rd** [reply]
Ah nice, good work! I guess there's something internal in alpha that changes things that image.nvim doesn't catch.
Maybe there's a way to hook after alpha's rerenders to draw the image again, that should fix it.
